sklearn/naive_bayes.py

```diff
@@ -958,3 +998,256 @@ def _joint_log_likelihood(self, X):
+            assumed to be from a different categorical distribution.
+
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        Returns
+        -------
+        self : object
+        """
+        return super(CategoricalNB, self).fit(X, y, sample_weight=None)
+
+    def partial_fit(self, X, y, classes=None):
+        """Incremental fit on a batch of samples.
+
+        This method is expected to be called several times consecutively
+        on different chunks of a dataset so as to implement out-of-core
+        or online learning.
+
+        This is especially useful when the whole dataset is too big to fit in
+        memory at once.
+
+        This method has some performance overhead hence it is better to call
+        partial_fit on chunks of data that are as large as possible
+        (as long as fitting in the memory budget) to hide the overhead.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features. Here, each feature of X is
+            assumed to be from a different categorical distribution.
+
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        classes : array-like, shape = [n_classes] (default=None)
+            List of all the classes that can possibly appear in the y vector.
+
+            Must be provided at the first call to partial_fit, can be omitted
+            in subsequent calls.
+
+        Returns
+        -------
+        self : object
+        """
+        return super(CategoricalNB, self).partial_fit(X, y, classes,
+                                                      sample_weight=None)
+
+    def _check_input(self, X, Y=None):
+        if Y is None:
+            if _old_numpy:
+                X = check_array(X, accept_sparse=False)
+                if not np.isfinite(X).all():
+                    raise ValueError("Input contains NaN or infinity.")
+                return np.asarray(X, dtype=np.int64)
+            else:
+                return check_array(X, accept_sparse=False)
+        else:
+            if _old_numpy:
+                X, Y = check_X_y(X, Y, accept_sparse=False)
+                if not np.isfinite(X).all():
+                    raise ValueError("Input contains NaN or infinity.")
+                X = np.asarray(X, dtype=np.int64)
+                return X, Y
+            else:
+                return check_X_y(X, Y, accept_sparse=False)
+
+    def _init_counters(self, n_effective_classes, n_features):
+        self.class_count_ = np.zeros(n_effective_classes, dtype=np.float64)
+        self.category_count_ = [np.zeros((self.class_count_.shape[0], 0))
+                                for _ in range(n_features)]
+        self.feature_cat_index_mapping_ = [{} for _ in range(n_features)]
```

Comment:
this attribute is not documented, nor its structure asserted in tests

Reply:
added tests and docstring